docs/guide/config_dynamic.md

```diff
@@ -44,6 +44,16 @@ $myParamSettingAdmin = $configResolver->getParameter( 'my_param', 'myapp', 'admi
 // $myParamSetting's value will be 'another value'
 ```
 
+In debug mode, ConfigResolver detects if parameters were loaded prior to initialization of SiteAccess to warn about issues that are hard to resolve.
+It will log all instances of `getParameter()` that may be used unsafely.
+
+To avoid issues:
+- Avoid eager usage of config resolver (e.g. in service factories).
+- Instead of using `ctor('$dynamic_param$')`, use `(setter('$dynamic_param$'))` as it allows the system to update the scope changes.
+- Load the parameter lazily by injecting ConfigResolver with the required value only when necessary.
```

Comment:
```suggestion
- Load the parameter lazily by injecting ConfigResolver, and get the parameter from it _when_ you need to instead of during construction.
```